lib/Backend/BackwardPass.cpp

```diff
@@ -3244,7 +3244,15 @@ BackwardPass::ProcessBlock(BasicBlock * block)
                     FOREACH_SLIST_ENTRY(IR::ByteCodeUsesInstr*, bcuInstr, loop->outwardSpeculationMaskInstrs)
                     {
                         // Get the upwardExposed information for the previous block
-                        IR::LabelInstr *blockLabel = bcuInstr->m_prev->AsLabelInstr();
+                        IR::LabelInstr *blockLabel = nullptr;
+                        {
+                            IR::Instr* temp = bcuInstr->m_prev;
+                            while (!temp->StartsBasicBlock())
+                            {
+                                temp = temp->m_prev;
+                            }
+                            blockLabel = temp->AsLabelInstr();
+                        }
                         BasicBlock* maskingBlock = blockLabel->GetBasicBlock();
```

Comment:
this could be a new function GetPrevLabelInstr ? There might be other places we might need this as well.

Reply:
Good point; I've made two new functions for the next commit set, GetPrevLabelInstr and GetBlockStartInstr (since, although in this case all the blocks are started by LabelInstrs, they can theoretically be started by EntryInstrs).